nextcord/guild.py

```diff
@@ -2902,7 +2911,25 @@ async def ban(
         HTTPException
             Banning failed.
         """
-        await self._state.http.ban(user.id, self.id, delete_message_days, reason=reason)
+        if delete_message_days is not None and delete_message_seconds is not None:
+            raise InvalidArgument(
+                "Cannot pass both delete_message_days and delete_message_seconds."
+            )
+        elif delete_message_days is None and delete_message_seconds is None:
+            # Default to one day
+            delete_message_seconds = 24 * 60 * 60
+
+        if delete_message_days is not None:
+            warnings.warn(
+                DeprecationWarning(
+                    "delete_message_days is deprecated, use delete_message_seconds instead."
+                )
+            )
+            delete_message_seconds = delete_message_days * 24 * 60 * 60
+
+        await self._state.http.ban(
+            user.id, self.id, cast(int, delete_message_seconds), reason=reason
```

Comment:
Since casting an object doesn't actually do anything to the object itself at runtime, this is pretty unsafe if `delete_message_seconds` isn't passed into this method
First of all, it violates the type hints of the `ban` method and second of all I don't think Discord can handle `null` for that attribute

Reply:
Resolved via a Discord conversation starting at this message: https://canary.discord.com/channels/881118111967883295/881124672056467456/1018006566097666098